File: tasks/translation/output/Qwen/Qwen2.5-Coder-32B-Instruct-AWQ/codenet/cot/Java/C++/s936984238_commented.cpp
#include <iostream>
#include <vector>
#include <algorithm>

std::vector<int> score;
int number;

bool read() {
    std::cin >> number;
    if (number == 0)
        return false;
    
    score.resize(number);
    for (int i = 0; i < number; i++) {
        std::cin >> score[i];
    }
    return true;
}

void solve() {
    std::sort(score.begin(), score.end());
    int sum = 0;
    for (int i = 1; i < number - 1; i++) {
        sum += score[i];
    }
    std::cout << sum / (number - 2) << std::endl;
}

int main() {
    while (read()) {
        solve();
    }
}
// End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Go/s936984238_commented.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

var sc = bufio.NewScanner(os.Stdin)

func main() {
	for read() {
		solve()
	}
}

func read() bool {
	var number int
	fmt.Scan(&number)
	if number == 0 {
		return false
	}

	score := make([]int, number)
	for i := 0; i < number; i++ {
		fmt.Scan(&score[i])
	}
	return true
}

func solve() {
	sort.Ints(score)
	sum := 0
	for i := 1; i < number-1; i++ {
		sum += score[i]
	}
	fmt.Println(sum / (number - 2))
}

// End of Code
